pkg/server/preprocessor/auth.go

```diff
@@ -68,15 +68,23 @@ func (p AccessKeyValidationPreprocessor) Preprocess(payload *router.Payload, res
 
 // UserAuthenticator provides preprocess method to authenicate a user
 // with access token or non-login user without api key.
+// if ByPassUnauthorized is true, UserAuthenticator will only inject AuthInfoID
+// and related context, and always return StatusOK
+// it is for plugin so even handler or lambda are not user_required,
+// we can still get the user id from context
```

Comment:
See if this is better?

```
// UserAuthenticator provides preprocess method to authenicate a user
// with access token or non-login user without api key.
// It inject AuthInfoID and related context.
// If BypassUnauthorized is true, UserAuthenticator will return
// StatusOK instead of StatusUnauthorized if the request is not
// authenticated. It is for plugin so even handler or lambda
// that are not user_required, we can still get the
// AuthInfoID from context.
```